dashboard/views.py

```diff
@@ -429,10 +431,22 @@ def resource_access_within_week(request, course_id=0):
         axis=1)
     # RESOURCE_VALUES_MAP {'canvas': 'files', 'leccap': 'videos', 'mivideo': 'videos'}
     output_df['resource_type'] = output_df['resource_type'].replace(RESOURCE_VALUES_MAP)
+ 
+     # Limit the number of results for large courses
+    total_rows = df.shape[0]
+    if total_rows > config.RESOURCE_LIMIT:
+        output_df = output_df.sort_values("total_percent", ascending=False).head(config.RESOURCE_LIMIT)
     output_df.drop(columns=['name', 'resource_id_type'], inplace=True)
 
     logger.debug(output_df.to_json(orient='records'))
-    return HttpResponse(output_df.to_json(orient='records'),content_type='application/json')
+    response = HttpResponse(output_df.to_json(orient='records'), content_type='application/json')
+
+    # Add in this header if needed to pass that we limited this to the frontend
+    if total_rows > config.RESOURCE_LIMIT:
+        response['Resources-Limit'] = config.RESOURCE_LIMIT
+        response['Access-Control-Expose-Headers'] = 'Resources-Limit'
```

Comment:
Is this standard practice for something like this? Maybe we need the state and other changes in the frontend then, rather than going with the `globals` approach. Let me know what you think.